iron-form.html

```diff
@@ -95,6 +95,43 @@
   </template>
 
   <script>
+  (function() {
+    function isSubmittable(node, ignoreName) {
+      // An element is submittable if it is not disabled, and if it has a
+      // 'name' attribute
+      return  (!node.disabled && (ignoreName || node.name));
+    }
+
+    /**
+     * Traverse the distributed nodes of a slot or content element
+     * and add all submittable nodes to `submittable`.
+     * @param  {Array<Node>} submittable Reference to the arrray of submittables
+     * @param  {Node} node        The slot or content node
+     * @param  {boolean} ignoreName  Whether the name of the submittable nodes should be disregarded
+     * @return {void}
+     */
+    function handleSubmittableInSlot(submittable, node, ignoreName) {
+      var assignedNodes = Polymer.dom(node).getDistributedNodes();
+
+      for (var i = 0; i < assignedNodes.length; i++) {
+        if (assignedNodes[i].nodeType === Node.TEXT_NODE) {
+          continue;
+        }
+
+        if (isSubmittable(assignedNodes[i], ignoreName)) {
```

Comment:
I spent some time figuring out if we can finish at this spot with `assignedNodes[i]` being a `<slot>`, because if that is a named slot (e.g. `<slot name="inputs">`) this logic would consider it submittable :x
Fortunately `Polymer.dom.getDistributedNodes()` method will flatten all the slots and give us the actual elements 🎉 

nit: mind adding a comment at this line for the sanity of future maintainers saying that `assignedNodes` does not contain `<slot>` or `<content>` because `getDistributedNodes` flattens?

Reply:
FYI this is my little test for checking 2-level distribution:

```html
<link rel="import" href="iron-form.html">

<slotted-slotted-form id="slottedSlottedForm">
  <input slot="inputs" name="lightDomInput" value="in light dom" />
</slotted-slotted-form>

<dom-module id="slotted-slotted-form">
  <template>
    <slotted-form id="slottedForm">
      <input slot="inputs" name="slottedSlottedInput" value="in slotted slotted" />
      <slot slot="inputs" name="inputs"></slot>
    </slotted-form>
  </template>
  <script>
    Polymer({
      is: 'slotted-slotted-form'
    });
  </script>
</dom-module>

<dom-module id="slotted-form">
  <template>
    <iron-form id="form">
      <form>
        <input slot="inputs" name="slottedInput" value="in slotted" />
        <slot name="inputs"></slot>
      </form>
    </iron-form>
  </template>
  <script>
    Polymer({
      is: 'slotted-form',
      attached: function() {
        // Wait for observeNodes.
        this.async(() => console.log(this.$.form._getValidatableElements()));
      }
    });
  </script>
</dom-module>

Reply:
I'll add this comment on a followup PR so that we can merge :)